docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+
+1. Receive message archive index signal, download index, then determine which message archives to download
+3. Download individual archives
+
+Community member nodes subscribe to the read-only channel that community owner nodes publish message archive indices to. There are two scenarios in which member nodes can receive message archive index messages:
+
+1. The member node receives it via live messages, that is, messages that are relayed by store nodes 
+2. The member node requests messages for a time range of up to 30 days from store nodes (this is the case when a new community member joins a community)
+
+### Downloading message archive indices
+When member nodes receive a message with a `CommunityMessageArchiveIndex` ([6/PAYLOADS](/specs/6-payloads)) from the aforementioned channnel, they MUST extract the `magnet_uri` and pass it to their underlying BitTorrent client so they can fetch the latest message archive index.
+
+Due to the nature of distributed systems, there's no guarantee that a received message is the "last" message. This is especially true when member nodes request historical messages from store nodes. 
+
+Therefore, member nodes MUST wait for 20 seconds after receiving the last `CommunityMessageArchiveIndex` before they start extracting the magnet link to fetch the latest archive index.
+
+### Downloading individual archives
+Once a message archive index is downloaded, community member nodes use a local lookup table to determine which of the listed archives are missing. For this lookup to work, member nodes MUST store the KECCAK-256 hashes of the magnet links for archives they've downloaded.
+
+Given a list of magnet links provided by the index to download individual archives, member nodes can:
+
+1. **Download all archives** - Extract each magnet link in the index and pass them to the underlying BitTorrent client (this is the case for new community member nodes that haven't downloaded any archives yet)
+2. **Download only the latest archive** - Extract only the newest magnet link and pass it to the BitTorrent client (this the case for any member node that already has downloaded all previous history and is now interested in only the latst archive)
+3. **Download specific archives** - Look into `from` and `to` fields of every `WakuMessageArchiveMetadata` and only extract magnet links for archives of a specific time range (can be the case for member nodes that have recently joined the network and are only interested in a subset of the complete history)
+
+
+## Storing historical messages
+
+When message archives are fetched, community member nodes MUST unwrap the resulting `WakuMessage` instances into `ApplicationMetadataMessage` instances and store them in their local database.
+Community member nodes SHOULD NOT store the wrapped `WakuMessage` messages.
+
+Already stored messages with the same `id` or `clock` value MUST be replaced with messages extracted from archives, if both of these values are equal.
```

Comment:
@staheri14 I know we talked about this but I still went for "only replace what needs replacement" for now. This can cause inconsistency compared to the canonical history though, so if we really want to replace everything from T1  - T2, no matter what, let me know and I'll update this.

/cc @John-44

Reply:
I also agree with the consistency i.e., replacing everything from T1-T2, we can later design a synchronization protocol across store nodes to make sure they all have consistent message hisotry